sematic/scheduling/kubernetes.py

```diff
@@ -517,41 +286,217 @@ def refresh_job(job: ExternalJob) -> KubernetesExternalJob:
+            phase=pod.status.phase,
+            condition_message=most_recent_condition_message,
+            condition=most_recent_condition_name,
+            container_condition_message=most_recent_container_condition_message,
+            container_exit_code=container_exit_code,
+            start_time_epoch_seconds=(
+                pod.status.start_time.timestamp()
+                if pod.status.start_time is not None
+                else None
+            ),
+            has_infra_failure=has_infra_failure,
+            unschedulable_message=unschedulable_reason,
+            node_name=node_name,
+        )
+    except Exception as e:
+        pod_name = "Unknown"
+        try:
+            pod_name = pod.metadata.name  # type: ignore
+        except Exception as e2:
+            logger.error("Pod name could not be determined", exc_info=e2)
+        logger.error(
+            "Got exception while extracting information from pods: %s",
+            pod_name,
+            exc_info=e,
+        )
+        return PodSummary(
+            pod_name=pod_name,
+            has_infra_failure=True,
+        )
+
+
+def _get_pods_for_job(
+    job: Job,
+) -> Tuple[Optional[List[V1Pod]], bool]:
+    """Get the pod(s) associated with the given job
+
+    Parameters
+    ----------
+    job:
+        The job to get pods for
+
+    Returns
+    -------
+    A tuple where the first element is the list of pods, if the list can
+    be determined. If the list of pods can't be determined due to error,
+    the first element will be None.
+    The second element of the tuple is a boolean indicating if there was
+    an infra failure during pod retrieval (True if there was an error).
+    """
+    k8s_pods = None
+    has_infra_failure = False
+    try:
+        k8s_pods = kubernetes.client.CoreV1Api().list_namespaced_pod(
+            namespace=job.namespace,
+            label_selector=f"job-name={job.name}",
+        )
+        logger.debug("K8 pods for job %s/%s:\n%s", job.namespace, job.name, k8s_pods)
+
+        if _is_none_or_empty(k8s_pods.items):
+            return [], has_infra_failure
+        return list(k8s_pods.items), has_infra_failure
+    except OpenApiException as e:
+        has_infra_failure = True
+        logger.warning(
+            "Got exception while looking for pods for job %s/%s",
+            job.namespace,
+            job.name,
+            exc_info=e,
+        )
+        return None, has_infra_failure
+
+
+def _get_unschedulable_reason(pod_conditions) -> Optional[str]:
+    message = None
+    for condition in pod_conditions:
+        if condition.type != "PodScheduled" or condition.status == "True":
+            continue
+
+        message = (
+            f"Pod is not scheduled. Reason: {condition.reason}: {condition.message}. "
+            f"Depending on your Kubernetes cluster's configuration and current usage, "
+            f"this may or may not resolve itself on its own. Please consult your "
```

Comment:
Please add clarifications:
```suggestion
            f"this may or may not resolve itself on its own through autoscaling or resource reclaiming. Please consult your "
```